lib/Migration/Version5000Date20211025124248.php

```diff
@@ -0,0 +1,218 @@
+			]);
+			$table->addColumn('configuration', Types::TEXT, [
+				'notnull' => true,
+			]);
+			$table->setPrimaryKey(['id'], 'idx_user_saml_config');
+		}
+
+		return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		$prefixes = $this->fetchPrefixes();
+		foreach ($prefixes as $prefix) {
+			$keyStart = $prefix === 1 ? '' : $prefix . '-';
+			$configKeys = array_reduce(
+				self::IDP_CONFIG_KEYS,
+				function (array $carry, string $rawConfigKey) use ($keyStart): array {
+					$carry[] = $keyStart . $rawConfigKey;
+					return $carry;
+				},
+				[]
+			);
+
+			$configData = $keysToDelete = [];
+			$gRows = $this->readConfiguration($configKeys);
+			while ($row = $gRows->current()) {
+				$configData[$this->normalizeKey($row['configkey'])] = $row['configvalue'];
+				$keysToDelete[] = $row['configkey'];
+				$gRows->next();
+			}
+
+			if ($this->insertConfiguration($prefix, $configData) && !empty($keysToDelete)) {
+				$this->deleteOldConfiguration($keysToDelete);
+			}
+		}
+
+		$this->deletePrefixes();
+	}
+
+	protected function deleteOldConfiguration(array $keys): bool {
+		static $deleteQuery;
+		if (!$deleteQuery) {
+			$deleteQuery = $this->dbc->getQueryBuilder();
+
+			$deleteQuery->delete('appconfig')
+				->where($deleteQuery->expr()->eq('appid', $deleteQuery->createNamedParameter('user_saml')))
+				->andWhere($deleteQuery->expr()->in('configkey', $deleteQuery->createParameter('cfgKeys')));
+		}
+
+		$deletedRows = $deleteQuery
+			->setParameter('cfgKeys', $keys, IQueryBuilder::PARAM_STR_ARRAY)
+			->executeStatement();
+
+		return $deletedRows > 0;
+	}
+
+	protected function insertConfiguration(int $id, array $configData): bool {
+		static $insertQuery;
+		if (!$insertQuery) {
+			$insertQuery = $this->dbc->getQueryBuilder();
+			$insertQuery->insert('user_saml_configurations')
+				->values([
+					'id' => $insertQuery->createParameter('configId'),
+					'name' => $insertQuery->createParameter('displayName'),
+					'configuration' => $insertQuery->createParameter('configuration'),
+				]);
+		}
+
+		$insertedRows = $insertQuery
+			->setParameter('configId', $id)
+			->setParameter('displayName', $configData['general-idp0_display_name'] ?? '')
+			->setParameter('configuration', \json_encode($configData, JSON_THROW_ON_ERROR))
+			->executeStatement();
+
+		return $insertedRows > 0;
+	}
+
+	protected function readConfiguration(array $configKeys): \Generator {
+		static $readQuery;
+		if (!$readQuery) {
+			$readQuery = $this->dbc->getQueryBuilder();
+		}
+		$readQuery->select('configkey', 'configvalue')
+			->from('appconfig')
+			->where($readQuery->expr()->eq('appid', $readQuery->createNamedParameter('user_saml')))
+			->andWhere($readQuery->expr()->in('configkey', $readQuery->createParameter('cfgKeys')));
```

Comment:
assuming less than 1000 entries, otherwise this needs chunking

Reply:
There is no way to hit the limitation since $configKeys is just the list of IDP_CONFIG_KEYS with a one prefix.